Add support for Python 3.9

hugovk opened the pull request:
Add to tests, Trove classifiers and `docs/contributing.rst`.

Omitted from the `types` test because the lxml dependency isn't yet ready for 3.9.

bhrutledge reviewed (changes requested):
Thanks, @hugovk! You mentioned updating the trove classifiers, but I don't see [setup.cfg](https://github.com/pypa/twine/blob/master/setup.cfg) in this PR. Can you add that?

bhrutledge commented:
> Omitted from the `types` test because the lxml dependency isn't yet ready for 3.9.

I think it'd be helpful to add something like `#TODO: Add 3.9 when lxml supports it` above `types.strategy.matrix.python`, along with a link to a relevant issue if you have one.

hugovk commented:
Oops, classifier now added!

Here's the actual error:

```
Collecting mypy
  Downloading mypy-0.790-py3-none-any.whl (2.4 MB)
Collecting lxml
  Downloading lxml-4.5.2.tar.gz (4.5 MB)
    ERROR: Command errored out with exit status 1:
     command: /home/runner/work/twine/twine/.tox/types/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3mbjm8h6/lxml/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3mbjm8h6/lxml/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-3d1hc8ne
         cwd: /tmp/pip-install-3mbjm8h6/lxml/
    Complete output (3 lines):
    Building lxml version 4.5.2.
    Building without Cython.
    Error: Please make sure the libxml2 and libxslt development packages are installed.
```

https://github.com/hugovk/twine/runs/1254845165?check_suite_focus=true

https://lxml.de/installation.html confirms that libxml2 and libxslt are needed. But perhaps it's easier to wait for wheels. I opened https://bugs.launchpad.net/lxml/+bug/1899830 to request them, and added it in the TODO.

hugovk commented:
lxml wheels are now available, added 3.9 typing tests and removed the TODO!

bhrutledge reviewed (approved):
Thanks!
